### ISSUE TITLE
Suggestion: Allow partial response mocking

### ISSUE THREAD

**Issue by JoshWhite** [miragejs/discuss#40]
I've just had a go at chucking this into a mature app, with an existing API.
I only need to make a small change; eg. add a couple of properties, to an existing call.

But I want the rest of the api response to remain the same. 

It would be great if you could trap a response from the server & manipulate it before handing back to the calling code.

**Comment by samselikoff** [comment]
Interesting idea!

Mirage route handlers are promise-aware so you could try fetching from your Mirage code, massaging and the responding. (Then it wouldn't work offline/locally.)

I've also seen folks copy+paste an existing response into their repo and using the [passthrough](https://miragejs.com/api/classes/server#passthrough) api to let all other responses hit the existing API.

**Comment by JoshWhite** [comment]
Yeah, I’m using the passthrough api 👍.

Ah ok, you mean hijack the request header in a route function, hand the call onto the api, then modify its response before returning?

Sounds like you can already do this then 😀

**Comment by samselikoff** [comment]
Yes, but re-caking the api might involve a bit of boilerplate. Basically there’s nothing first class in Mirage to support this but because route handlers wait on promises you might be able to do it today!
